### PR TITLE
Eager HashedCollections Bridging

### PR DESCRIPTION
**Pull request by Gankra**
Now that the rewrite of HashedCollections is done, we can move forward with looking at the impact of eager bridging Dictionary and Set. Not implemented in this PR is the refactoring of the design of bridging that @dabrahams did. That was just to get this out tonight to run benching/testing and get some feedback.

### CODE REVIEW

**Comment by Gankro** [comment]
@swift-ci Please smoke test

**Comment by Gankro** [comment]
@swift-ci Please benchmark

File: validation-test/stdlib/Dictionary.swift [hunk 1]
```diff
@@ -1312,6 +1312,30 @@ class ParallelArrayDictionary : NSDictionary {
...
+      switch idx {
+      case 1: return dict.keys[0].key0
+      case 2: return dict.keys[0].key1
+      case 3: return dict.keys[0].key2
+      case 4: return dict.keys[0].key3
+      default: return nil
+      }
+    }
+  }
 }
```
Comment (Gankro) [inline validation-test/stdlib/Dictionary.swift]: Note to self: upon reflection this fix is dubious; it implies the motivation for this type is no longer happening. Need to investigate what the point of this *really* is.

Also note: the existing implementation appears to also have UB (strict aliasing violation).

**Comment by Gankro** [comment]
@swift-ci please smoke test

**Comment by Gankro** [comment]
rebased and added updates to reflection tests. Annoyingly, updating the size of a type means all the offsets get messed up.

**Comment by Gankro** [comment]
@slavapestov is verifying the offsets (which are naturally just the sum of strides) really interesting?

**Comment by slavapestov** [comment]
Hi @Gankro can you hold off landing this until I land some resilience fixes? Your previous changes broke the resilient build and I'm still debugging them.

**Comment by Gankro** [comment]
@slavapestov this won't land for a long while; these branches are mostly here so everyone who wants to evaluate their consequences can.


Also: dang I'm really sorry about constantly breakinn resilience :(

**Comment by slavapestov** [comment]
No worries, it's good to shake all the bugs out.

**Comment by CodaFi** [comment]
It looks like we never did get around to eager bridging.  Should this remain open @Gankra?

**Comment by Gankra** [comment]
i feel like y'all should maybe have a better understanding of what your project should do than me, someone who hasn't worked here for 3 years?

**Comment by CodaFi** [comment]
@Gankra Very well.  Thank you for all the work that went into this patch.  I'm just sorry we weren't able to take it at the time.

I've spoken to the stdlib folk and they believe there may still be limited cases where this optimization makes sense.  I've filed [SR-11805](https://bugs.swift.org/browse/SR-11805) to see if an external contributor would like to take up the mantle at some point in the future.  With luck, your patch may yet make it through!

**Pull request closed without merging**